From 94abc76cd88ec9edcdf611c3587fae5763cde80e Mon Sep 17 00:00:00 2001
From: Fabio Valentini <decathorpe@gmail.com>
Date: Fri, 17 Sep 2021 14:46:32 +0200
Subject: [PATCH 2/2] mark flatpak sources in the UI

---
 src/Core/Package.vala | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/src/Core/Package.vala b/src/Core/Package.vala
index dcaed31a..4cfd2f5a 100644
--- a/src/Core/Package.vala
+++ b/src/Core/Package.vala
@@ -375,11 +375,14 @@ public class AppCenterCore.Package : Object {
                 }
             } else if (backend is FlatpakBackend) {
                 var fp_package = this as FlatpakPackage;
-                if (fp_package != null && fp_package.installation == FlatpakBackend.system_installation) {
-                    return _("%s (system-wide)").printf (origin);
+
+                if (fp_package != null && fp_package.installation != FlatpakBackend.system_installation) {
+                    return "%s (flatpak)".printf (origin);
+                } else if (fp_package != null && fp_package.installation == FlatpakBackend.system_installation) {
+                    return _("%s (system-wide)").printf (origin).replace (")", " flatpak)");
                 }
 
-                return origin;
+                return "%s (flatpak)".printf (origin);
             }
 
             return _("Unknown Origin (non-curated)");
-- 
2.34.1

